Tests: timeout in 'reload' test case

```python
[gw1] [ 11%] FAILED tests/test_herbstluftwm.py::test_reload 
=================================== FAILURES ===================================
_________________________________ test_reload __________________________________
[gw1] linux -- Python 3.8.2 /hlwm/.tox/py38/bin/python
hlwm_process = <conftest.HlwmProcess object at 0x7fc877addee0>
hlwm = <conftest.HlwmBridge object at 0x7fc877add490>
    def test_reload(hlwm_process, hlwm):
        with hlwm_process.wait_stdout_match('hlwm started'):
>           hlwm.call('reload')
tests/test_herbstluftwm.py:15: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/lib/python3.8/contextlib.py:120: in __exit__
    next(self.gen)
tests/conftest.py:400: in wait_stdout_match
    self.read_and_echo_output(until_stdout=match)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <conftest.HlwmProcess object at 0x7fc877addee0>
until_stdout = 'hlwm started', until_stderr = None, until_eof = False
    def read_and_echo_output(self, until_stdout=None, until_stderr=None, until_eof=False):
        expect_sth = ((until_stdout or until_stderr) is not None)
        max_wait = 15
    
        # Track which file objects have EOFed:
        eof_fileobjs = set()
        fileobjs = set(k.fileobj for k in self.output_selector.get_map().values())
    
        stderr = ''
        stderr_bytes = bytes()
        stdout = ''
        stdout_bytes = bytes()
    
        def match_found():
            if until_stdout and (until_stdout in stdout):
                return True
            if until_stderr and (until_stderr in stderr):
                return True
            return False
    
        started = datetime.now()
        while (datetime.now() - started).total_seconds() < max_wait:
            select_timeout = 1
            # If we're not polling for a matching string (anymore), there is no
            # need for a dampening timeout:
            if not expect_sth or match_found():
                select_timeout = 0
            selected = self.output_selector.select(timeout=select_timeout)
            for key, events in selected:
                # Read only single byte, otherwise we might block:
                ch = key.fileobj.read(1)
    
                if ch == b'':
                    eof_fileobjs.add(key.fileobj)
    
                # Store in temporary buffer for string matching:
                if key.fileobj == self.proc.stderr:
                    stderr_bytes += ch
                    if ch == b'\n':
                        stderr += stderr_bytes.decode()
                        # Pass it through to the real stderr:
                        key.data.write(stderr_bytes.decode())
                        key.data.flush()
                        stderr_bytes = b''
    
                if key.fileobj == self.proc.stdout:
                    stdout_bytes += ch
                    if ch == b'\n':
                        stdout += stdout_bytes.decode()
                        # Pass it through to the real stdout:
                        key.data.write(stdout_bytes.decode())
                        key.data.flush()
                        stdout_bytes = b''
    
            if until_eof:
                # We are going to the very end, so carry on until all file
                # objects have returned EOF:
                if eof_fileobjs == fileobjs:
                    break
                else:
                    continue
    
            if selected != []:
                # There is still data available, so keep reading (no matter
                # what):
                continue
    
            # But stop reading if there is nothing to look for or we have
            # already found it:
            if not expect_sth or match_found():
                break
    
        duration = (datetime.now() - started).total_seconds()
        if expect_sth and not match_found():
>           assert False, f'Expected string not encountered within {duration:.1f} seconds'
E           AssertionError: Expected string not encountered within 15.0 seconds
E           assert False
tests/conftest.py:390: AssertionError
---------------------------- Captured stdout setup -----------------------------
hlwm started
---------------------------- Captured stderr setup -----------------------------
EWMH: 1 desktops: 'default'
0 managed clients: 
----------------------------- Captured stdout call -----------------------------
Client command ['reload'] succeeded (no output)
hlwm started
=========================== short test summary info ============================
FAILED tests/test_herbstluftwm.py::test_reload - AssertionError: Expected str...
!!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!
!!!!!!!!!!!! xdist.dsession.Interrupted: stopping after 1 failures !!!!!!!!!!!!!
================== 1 failed, 128 passed in 202.69s (0:03:22) ===================
ERROR: InvocationError for command /hlwm/.tox/py38/bin/python -m pytest -n auto --cache-clear -v -x -m 'not exclude_from_coverage' (exited with code 2)
___________________________________ summary ____________________________________
ERROR:   py38: commands failed
Traceback (most recent call last):
  File "/hlwm/ci/build.py", line 132, in <module>
    tox('-e py38 -- -n auto --cache-clear -v -x -m "not exclude_from_coverage"', build_dir)
  File "/hlwm/ci/build.py", line 17, in tox
    sp.check_call(f'tox {tox_args}', shell=True, cwd=build_dir, env=tox_env)
  File "/usr/lib/python3.8/subprocess.py", line 364, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command 'tox -e py38 -- -n auto --cache-clear -v -x -m "not exclude_from_coverage"' returned non-zero exit status 1.
The command "docker run -u $UID --privileged -t --rm --volume=/etc/passwd:/etc/passwd:ro --volume=$HOME/.ccache-focal-gcc:/.ccache:rw --volume=$PWD:/hlwm:rw --volume=$HOME/.tox-cache:/hlwm/.tox:rw focal sh -c ' /hlwm/ci/build.py --build-dir=/hlwm/build --cxx=g++-9 --cc=gcc-9 --build-type=Debug --ccache=/.ccache --compile --run-tests '" exited with 1.
```

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -385,6 +385,15 @@ class HlwmProcess:
             if not expect_sth or match_found():
                 break
 
+        # decode remaining bytes for the final match_found() check
+        if stderr_bytes != b'':
+            stderr += stderr_bytes.decode()
+            sys.stderr.write(stderr_bytes.decode())
+            sys.stderr.flush()
+        if stdout_bytes != b'':
+            stdout += stdout_bytes.decode()
+            sys.stdout.write(stdout_bytes.decode())
+            sys.stdout.flush()
         duration = (datetime.now() - started).total_seconds()
         if expect_sth and not match_found():
             assert False, f'Expected string not encountered within {duration:.1f} seconds'
@@ -394,6 +403,9 @@ class HlwmProcess:
         """
         Context manager for wrapping commands that are expected to result in
         certain output on hlwm's stdout (e.g., input events).
+
+        Warning: do not run call(...) within such a context, but only
+        unchecked_call(..., read_hlwm_output=False) instead
         """
         self.read_and_echo_output()
         yield
@@ -404,6 +416,9 @@ class HlwmProcess:
         """
         Context manager for wrapping commands that are expected to result in
         certain output on hlwm's stderr (e.g., input events).
+
+        Warning: do not run call(...) within such a context, but only
+        unchecked_call(..., read_hlwm_output=False) instead
         """
         self.read_and_echo_output()
         yield

--- a/tests/test_herbstluftwm.py
+++ b/tests/test_herbstluftwm.py
@@ -12,7 +12,12 @@ HLWM_PATH = os.path.join(BINDIR, 'herbstluftwm')
 
 def test_reload(hlwm_process, hlwm):
     with hlwm_process.wait_stdout_match('hlwm started'):
-        hlwm.call('reload')
+        # run the command, but read not hlwm's output in unchecked_call()
+        # but instead, let the current context manager read it!
+        proc = hlwm.unchecked_call('reload', read_hlwm_output=False)
+        assert not proc.stderr
+        assert not proc.stdout
+        assert proc.returncode == 0
 
 
 def test_herbstluftwm_already_running(hlwm):

--- a/tests/test_spawn.py
+++ b/tests/test_spawn.py
@@ -1,4 +1,7 @@
 def test_spawn(hlwm, hlwm_process):
-    hlwm.call(['spawn', 'sh', '-c', 'echo >&2 spawnyboi'])
-
-    hlwm_process.wait_stderr_match('spawnyboi')
+    with hlwm_process.wait_stderr_match('spawnyboi'):
+        cmd = ['spawn', 'sh', '-c', 'echo >&2 spawnyboi']
+        proc = hlwm.unchecked_call(cmd, read_hlwm_output=False)
+        assert proc.returncode == 0
+        assert not proc.stderr
+        assert not proc.stdout

